portal/src/graphql/adminapi/AuditLogScreen.tsx

```diff
@@ -81,13 +92,24 @@ function RefreshButton(props: ICommandBarItemProps) {
   );
 }
 
+// eslint-disable-next-line complexity
 const AuditLogScreen: React.FC = function AuditLogScreen() {
-  const [offset, setOffset] = useState(0);
-  const [selectedKey, setSelectedKey] = useState("ALL");
+  const [searchParams, setSearchParams] = useSearchParams();
+
+  const fromParam = searchParams.get("from");
+  const toParam = searchParams.get("to");
+  const orderParam =
+    searchParams.get("order_by") === SortDirection.Asc
+      ? SortDirection.Asc
+      : SortDirection.Desc;
+  const pageParam = Number(searchParams.get("page"));
+  const offsetParam = pageParam > 1 ? (Number(pageParam) - 1) * pageSize : 0;
+  const selectedKeyParam = searchParams.get("activity_type");
+
+  const [offset, setOffset] = useState(offsetParam);
+  const [selectedKey, setSelectedKey] = useState(selectedKeyParam ?? "ALL");
   const [dateRangeDialogHidden, setDateRangeDialogHidden] = useState(true);
-  const [sortDirection, setSortDirection] = useState<SortDirection>(
-    SortDirection.Desc
-  );
+  const [sortDirection, setSortDirection] = useState<SortDirection>(orderParam);
   const [lastUpdatedAt, setLastUpdatedAt] = useState(new Date());
```

Comment:
We need to persist this as well. Otherwise, when the log page is visited again, there could be new items being prepended to the front of the list. Persist this in numeric format, i.e. the milliseconds from the epoch.